### PR TITLE
Add reference for Factory Defintion and Container Secrets

### PR DESCRIPTION
**Pull request by doanac**

### CODE REVIEW

**Comment by doanac** [comment]
Docs for 3a1333065c8ef83bf31607f2d176b44bc625ff3b are browesable at: https://ci.foundries.io/projects/fio-docs/builds/384/docs/artifacts/html/index.html

**Comment by doanac** [comment]
These always get rendered a little awkward but here are the two new links:

 https://ci.foundries.io/projects/fio-docs/builds/384/docs/artifacts/html/reference/container-secrets.html
 https://ci.foundries.io/projects/fio-docs/builds/384/docs/artifacts/html/reference/factory-definition.html

File: source/reference/factory-definition.rst [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    #  refs/heads/master:
+    #    - tag: postmerge
+    #  refs/heads/devel-foundries:
+    #    - tag: devel
+    #  refs/heads/devel-foundries-base:
+    #    - tag: devel-base
+    #      inherit: devel
+
+
+  # Optional - This allows a factory to use a ci-scripts repo that might have
```
Comment (mike-scott) [inline source/reference/factory-definition.rst]: We might want to just leave this last part out.  Just my opinion tho.

Comment (doanac) [reply source/reference/factory-definition.rst]: That's a fair point. Its almost too advanced. @EmbeddedAndroid - do you agree?

Comment (doanac) [reply source/reference/factory-definition.rst]: I've removed this. I'm going to merge and publish. We can always add it back.

**Comment by doanac** [comment]
Docs for 32d827c92c08525804b7b72bf53bf1f5d297ce90 are browesable at: https://ci.foundries.io/projects/fio-docs/builds/385/docs/artifacts/html/index.html

**Pull request merged**